### PR TITLE
Test:  Render widget with testData

### PR DESCRIPTION
**Pull request by GordonSmith**
Test rendering each widget with a call to testData.
MorphText:  Added testData method.
Menu:  Fixed invalid shape property.
Widget:  Allow html-color of type null or ""
Edge:  Fix error on empty render.
Graph:  Allow layout of type "None".
Vertex:  Fixed color typo.
GMap:  Fixed Layout typo.

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@mlzummo Please Review.
@jbrundage Please Review (note:  the publish color test changes)
@anmoljagetia Please review (note:  the widget render test changes).

**Comment by anmoljagetia** [comment]
Looks good!

File: src/common/Widget.js [hunk 1]
```diff
@@ -171,13 +171,15 @@
...
                     break;
                 case "html-color":
-                    var litmus = 'red';
-                    var d = document.createElement('div');
-                    d.style.color=litmus;
-                    d.style.color=_;
-                    //Element's style.color will be reverted to litmus or set to '' if an invalid color is given
-                    if( _ !== litmus && (d.style.color === litmus || d.style.color === '')){
-                        console.log("Invalid value for '" + id + "':  " + _);
+                    if (window.__hpcc_debug && _ && _ !== "red") {
```
Comment (mlzummo) [inline src/common/Widget.js]: how come we are only running this code in debug mode now?

Comment (GordonSmith) [inline src/common/Widget.js]: IMO The test is too expensive to be used at runtime every time a color is set (specifically the creation of the DOM node).  This way we can enable the test in debug mode for the unit testing and / or other testing.  

Also at runtime it doesn't change the flow of the program, it just spits out a console message.

Comment (jbrundage) [inline src/common/Widget.js]: This makes sense. This parameter validation is really only useful to a developer (in debug mode) anyway.

**Pull request merged**